### ISSUE TITLE
Any plan on `3.3.4 Matching`

### ISSUE THREAD

**Issue by aztack** [koka-lang/koka#235]
[3.3.4 Matching](https://koka-lang.github.io/koka/doc/book.html#sec-matching) is still marked with `TODO`.
Is there any plan on finishing this section?

**Comment by daanx** [comment]
Yes :-)  but there is a lot to do. The samples directory contains many examples of matching if that helps

**Comment by TheLostLambda** [comment]
If nobody else gets to this first, I might give it a (draft) swing after looking through the examples, tests, and generally anywhere `match` appears in a `.kk` file! I reckon that would also be the place to mention destructuring / matching of parameters?
